bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-notifications.js

```diff
@@ -89,12 +90,417 @@ export default function defineOHBlocks_Notifications (f7, isGraalJs) {
+      this.appendValueInput('itemName')
+        .appendField('to item ')
+        .setCheck('oh_item')
+      this.setInputsInline(false)
+      this.setColour(0)
+      this.setTooltip('An action that sends a command to an item, like ON to KitchenLights')
+      this.setOutput(true, 'oh_notificationAction')
+
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-notifications.html#command-action')
+    }
+  }
+
+  javascriptGenerator.forBlock['oh_cloudNotification_commandAction'] = function (block) {
+    let command = javascriptGenerator.valueToCode(block, 'command', javascriptGenerator.ORDER_ATOMIC).replace(/'/g, '')
+    let itemName = javascriptGenerator.valueToCode(block, 'itemName', javascriptGenerator.ORDER_ATOMIC).replace(/'/g, '')
+    if (isGraalJs) {
+      return [`command:${itemName}:${command}`, javascriptGenerator.ORDER_ATOMIC]
+    } else {
+      throw new Error(unavailMsg)
+    }
+  }
+
+  Blockly.Blocks['oh_cloudNotification_uiAction'] = {
+    init: function () {
+      const actionImageField = new Blockly.FieldImage(actionImage, 15, 15, undefined)
+      this.appendDummyInput()
+        .appendField(actionImageField, 'actionHint')
+        .appendField('Call ')
+        .appendField(new Blockly.FieldDropdown([['Basic UI', 'ui:/basicui/app?'], ['UI absolute Path', 'ui:'], ['Main UI path', 'ui:navigate:'], ['Main UI popup', 'ui:popup:'], ['URL', '']]), 'command')
+      this.appendValueInput('path')
+        .appendField('with ')
+        .setCheck('String')
+      this.setInputsInline(false)
+      this.setColour(0)
+
+      this.setTooltip(() => {
+        const commandType = this.getFieldValue('command')
+        const TIP = {
+          'ui:/basicui/app?': 'Provide valid sitemap parameters in the form w=nnnn&sitemap=mysitemap',
+          'ui:': 'Provide the path that follows your openhab domain name, e.g /some/absolute/path like /page/rollershutter_firstfloor',
+          'ui:navigate:': '',
+          'ui:popup:': 'Provide the name of a page to be shown as a popup, e.g. page_phones',
+          '': 'Provide a valid absolute URL like https://openhab.org'
+        }
+        return TIP[commandType]
+      })
+      this.setOutput(true, 'oh_notificationAction')
+
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-notifications.html#ui-action')
+    }
+  }
+
+  javascriptGenerator.forBlock['oh_cloudNotification_uiAction'] = function (block) {
+    const command = block.getFieldValue('command')
+    let path = javascriptGenerator.valueToCode(block, 'path', javascriptGenerator.ORDER_ATOMIC).replace(/'/g, '')
+    if (isGraalJs) {
+      return [`${command}${path}`, javascriptGenerator.ORDER_ATOMIC]
+    } else {
+      throw new Error(unavailMsg)
+    }
+  }
+
+  Blockly.Blocks['oh_hideCloudNotification'] = {
+    init: function () {
+      this.appendValueInput('reference')
+        .appendField('hide notification by reference ')
```

Comment:
It's called reference ID:

```suggestion
        .appendField('hide notification by reference ID ')
```